src/views/credentials/addCredentialModal.tsx

```diff
@@ -52,283 +86,308 @@ const useCredentialForm = (credentialType: string | undefined, credential?: Cred
     };
-  }, [typeValue, credential]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   const handleInputChange = useCallback(
     (field: string, value: string) => {
       setFormData({ ...formData, [field]: value });
+      if (maskedFields[field]) {
+        setMaskedFields({ ...maskedFields, [field]: false });
+      }
     },
-    [formData]
+    [formData, maskedFields]
   );
 
-  const filterFormData = (data: CredentialFormType) =>
-    Object.fromEntries(Object.entries(data).filter(([, value]) => value));
+  const filterFormData = useCallback(() => {
+    const cleanedData = getCleanedFormData(
+      {
+        ...formData,
+        ...(!credential && { cred_type: typeValue }),
+        ...(credential && { id: credential.id })
+      },
+      authType
+    );
+
+    Object.keys(maskedFields).forEach(key => {
+      if (maskedFields[key]) {
+        delete cleanedData[key];
+      }
+    });
+
+    return cleanedData;
+  }, [authType, formData, credential, typeValue, maskedFields]);
 
   return {
     formData,
     authType,
     typeValue,
     setAuthType,
     handleInputChange,
-    filterFormData
+    filterFormData,
+    maskedFields
   };
 };
 
-const CredentialFormFields: React.FC<CredentialFormFieldsProps> = ({
-  formData,
-  authType = 'Username and Password',
-  typeValue = 'network',
-  setAuthType = Function.prototype,
-  handleInputChange = Function.prototype
-}) => (
-  <React.Fragment>
-    <FormGroup label="Name" isRequired fieldId="name">
-      <TextInput
-        value={formData?.name}
-        placeholder="Enter a name for the credential"
-        isRequired
-        type="text"
-        id="credential-name"
-        name="name"
-        onChange={event => handleInputChange('name', (event.target as HTMLInputElement).value)}
-        ouiaId="cred_name"
-      />
-    </FormGroup>
 
-    {/* Render Authentication Type dropdown only if needed based on the credential type */}
-    {(typeValue === 'network' || typeValue === 'openshift') && (
-      <FormGroup label="Authentication Type" fieldId="auth_type">
-        <SimpleDropdown
-          label={authType}
-          menuToggleOuiaId="auth_type"
-          variant="default"
-          isFullWidth
-          onSelect={item => setAuthType(item)}
-          dropdownItems={
-            (typeValue === 'network' && [
-              { item: 'Username and Password', ouiaId: 'password' },
-              { item: 'SSH Key', ouiaId: 'ssh_key' }
-            ]) || [
-              { item: 'Token', ouiaId: 'auth_token' },
-              { item: 'Username and Password', ouiaId: 'password' }
-            ]
-          }
-        />
-      </FormGroup>
-    )}
+const CredentialForm: React.FC<CredentialFormProps> = ({
+  credential,
+  credentialType,
+  onClose = () => {},
+  onSubmit = () => {},
+  useForm = useCredentialForm
+}) => {
+  const { formData, authType, typeValue, setAuthType, handleInputChange, filterFormData, maskedFields } = useForm({
```

Comment:
highly recommended 5:
we don't need to import `maskedFields` here `credential` has everything we need. remove `maskedFields` from `useForm`